### PR TITLE
Revise SDD for Svc::FramingProtocol

### PR DESCRIPTION
**Pull request by bocchino**
This PR updates the SDD for the Svc::FramingProtocol library.

* Revise overview
* Add requirements
* Document interface
* Document implementation

### CODE REVIEW

File: Svc/FramingProtocol/docs/sdd.md [hunk 1]
```diff
@@ -1,29 +1,314 @@
...
 
-FramingProtocol is used to build the messages for Framer and Deframer. It splits out the serialization from the topology hookup allowing the same system component connections to support different byte protocols "on-the-wire". F´ is shipped with a default "F´ protocol" which works with F´ Test Gds and amounts to: startword, data, CRC32.
+The `FramingProtocol` library defines the interfaces to the
+framing and deframing
+protocols used by the `Svc::Framer` and `Svc::Deframer` components.
+`Svc::Framer` uses a framing protocol to wrap data in frames
+for transmission to the ground.
+`Svc::Deframer` uses a deframing protocol to extract
+data from frames received from the ground.
+The `FramingProtocol` library allows each these components to
```
Comment (arizvi786) [inline Svc/FramingProtocol/docs/sdd.md:11]: allows each "of" these (missing 'of')

**Review by arizvi786 — COMMENTED** [review]
@bocchino looks good! Just one minor typo.

File: Svc/FramingProtocol/docs/sdd.md (hunk 1 above)
Comment (bocchino) [reply Svc/FramingProtocol/docs/sdd.md]: Good catch, fixed.

**Review by arizvi786 — APPROVED** [review]

**Review by kevin-f-ortega — APPROVED** [review]
LGTM

File: Svc/FramingProtocol/docs/sdd.md [hunk 2]
```diff
@@ -1,29 +1,314 @@
...
+
+1. Determine how many bytes are needed for deframing, peeking into
+the circular buffer if necessary (e.g., to read a length).
+
+1. If that many bytes are available in the circular buffer:
+
+   1. Use `m_interface->allocate` to allocate an `Fw::Buffer`
+to hold the deframed data.
+
+   1. Deframe the data into the allocated buffer.
```
Comment (LeStarch) [inline Svc/FramingProtocol/docs/sdd.md:235]: This should probably note that this is a copy.  Currently the calling component (Deframer) is expected to perform that actual consumption of the data.

File: Svc/FramingProtocol/docs/sdd.md [hunk 3]
```diff
@@ -1,29 +1,314 @@
...
+
+   1. Use `m_interface->route` to send the buffer.
+
+1. Return status.
```
Comment (LeStarch) [inline Svc/FramingProtocol/docs/sdd.md:239]: Need to ensure that the user sets the "needed" value correctly too.

File: Svc/FramingProtocol/docs/sdd.md [hunk 4]
```diff
@@ -1,29 +1,314 @@
...
 
-    virtual Fw::Time time() = 0;
-
     virtual void send(Fw::Buffer& outgoing) = 0;
 ```
 
-FramingProtocol Interface:
+The method `allocate` should accept a size in bytes and return (1) an `Fw::Buffer`
+that points to a memory allocation of at least that size if the allocation
+succeeded; or (2) an invalid buffer (i.e., a buffer whose address is zero)
```
Comment (LeStarch) [inline Svc/FramingProtocol/docs/sdd.md:130]: Is this true?  Or is it the size that is set to 0?

File: Svc/FramingProtocol/docs/sdd.md (hunk 2 above)
Comment (LeStarch) [reply Svc/FramingProtocol/docs/sdd.md:235]: This is, the caller rotates the circular buffer.

**Review by LeStarch — CHANGES_REQUESTED** [review]
Looks good, just a few nits.

File: Svc/FramingProtocol/docs/sdd.md (hunk 2 above)
Comment (bocchino) [reply Svc/FramingProtocol/docs/sdd.md:235]: Fixed.

File: Svc/FramingProtocol/docs/sdd.md (hunk 3 above)
Comment (bocchino) [reply Svc/FramingProtocol/docs/sdd.md:243]: Fixed.

**Review by LeStarch — APPROVED** [review]

**Pull request merged**